redhat-actions/try-in-web-ide
# Try to update previous comment instead of creating a new identical one

### Description



Updates the existing comment created by the try-in-web-ide github action instead of creating a new comment. The problem that this PR solves, is that when a PR is created when [add_comment](https://github.com/redhat-actions/try-in-web-ide/blob/d10b6215004b4cf04ab65f5ef3e7b98b32b40ca0/src/generated/inputs-outputs.ts#L8) is `true`, there are multiple new comments created by the github action if there are multiple [synchronize](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request) events (ie, if the PR branch has new changes, new commits etc.). 

This PR will try to update an existing comment that the GH action made, in case that the [comment_badge](https://github.com/redhat-actions/try-in-web-ide/blob/d10b6215004b4cf04ab65f5ef3e7b98b32b40ca0/src/generated/inputs-outputs.ts#L20) or the [web_ide_instance](https://github.com/redhat-actions/try-in-web-ide/blob/d10b6215004b4cf04ab65f5ef3e7b98b32b40ca0/src/generated/inputs-outputs.ts#L41) has been updated since the previous commit by the GH action was made. If the `comment_badge` and `web_ide_instance` are unchanged, there will be no update, and there will be no new comment.

### Related Issue(s)
https://github.com/redhat-actions/try-in-web-ide/issues/13

### Testing this PR
This PR has been tested in this repo: https://github.com/dkwon17/my-repo, [see](https://github.com/dkwon17/my-repo/blob/504127f46b7925352cfef2345b5225fec9611967/.github/workflows/try-in-web-ide.yaml#L15).

In https://github.com/dkwon17/my-repo/pull/8#issuecomment-1411082084, we can see that the comment has been updated by the github-actions bot with a new comment badge.

To test this on a personal repo:

1. Add this `./github/workflows/try-in-web-ide.yaml` file in your repo's main branch 
```
name: try-in-web-ide

on:
  pull_request_target: 
    types: [opened, synchronize]

jobs:
  add-link:
    runs-on: ubuntu-20.04
    steps:
      - name: "Checkout source code"
        uses: actions/checkout@v3
      - name: Eclipse Che Pull Request Link
        id: che-pr-check-gh-action
        uses: dkwon17/try-in-web-ide@update-comment
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          add_comment: true
          add_status: false
          web_ide_instance: https://workspaces.openshift.com
``` 

2. make any PR to your repo. This badge comment should be made by the github-actions bot:
<img width="924" alt="image" src="https://user-images.githubusercontent.com/83611742/215897715-08cfbc6e-7336-4e66-af54-0e2e61fd15db.png">

3. Make any change to the PR branch. There should be no new comment created by the github-actions bot.

4. Append the following to the `try-in-web-ide.yaml` file in the main branch:
```
          comment_badge: https://camo.githubusercontent.com/26774e52bd85d13b31d69c069e9c55939da6239a9011368563c09623d292ce29/68747470733a2f2f7777772e65636c697073652e6f72672f6368652f636f6e747269627574652e737667
```
5. Make another change to the PR branch. The comment created by the github-actions bot should be updated with the new comment badge.

### Checklist

- [ ] This PR includes a documentation change
- [ ] This PR does not need a documentation change
---
- [x] This PR includes test changes
- [ ] This PR's changes are already tested
---
- [ ] This change is not user-facing
- [ ] This change is a patch change
- [x] This change is a minor change
- [ ] This change is a major (breaking) change

### Changes made
 - Created new helper class called [UpdateCommentHelper](https://github.com/dkwon17/try-in-web-ide/blob/a43e6cf1febe4908a066eb5fbd45a142ef4e113c/src/helpers/update-comment-helper.ts), injected into the pull request handling logic: [handle-pull-request-logic.ts](https://github.com/redhat-actions/try-in-web-ide/compare/main...dkwon17:try-in-web-ide:update-comment?expand=1#diff-dd5d63a27a37671e2a8c9736b2620ac84ec3035d68958bce292368ea3ffedbeb)
 - Adds tests
